salt/modules/state.py

```diff
@@ -2319,21 +2321,21 @@ def event(tagmatch='*',
         salt-call --local state.event pretty=True
     '''
     sevent = salt.utils.event.get_event(
-            node,
-            sock_dir or __opts__['sock_dir'],
-            __opts__['transport'],
-            opts=__opts__,
-            listen=True)
+                 node,
+                 sock_dir or __opts__['sock_dir'],
+                 __opts__['transport'],
+                 opts=__opts__,
+                 listen=True)
 
     while True:
         ret = sevent.get_event(full=True, auto_reconnect=True)
         if ret is None:
             continue
 
-        if fnmatch.fnmatch(ret['tag'], tagmatch):
+        if salt.utils.stringutils.expr_match(ret['tag'], tagmatch):
             if not quiet:
                 salt.utils.stringutils.print_cli(
-                    str('{0}\t{1}').format(  # future lint: blacklisted-function
+                    '{0}\t{1}'.format(  # future lint: blacklisted-function
```

Comment:
Can you explain your reasoning for removing the `str()` here? It was done for a very good reason. With `unicode_literals` imported, and `json.dumps()` producing a `str` type, the format string has to also be a `str` type to avoid a decode error when the dumped data includes unicode characters:

```python
>>> from __future__ import unicode_literals
>>> import salt.utils.json
>>> data = {'спам': 'яйца'}
>>> data
{u'\u0441\u043f\u0430\u043c': u'\u044f\u0439\u0446\u0430'}
>>> '{0}'.format(salt.utils.json.dumps(data))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'ascii' codec can't decode byte 0xd1 in position 2: ordinal not in range(128)
>>> str('{0}').format(salt.utils.json.dumps(data))
'{"\xd1\x81\xd0\xbf\xd0\xb0\xd0\xbc": "\xd1\x8f\xd0\xb9\xd1\x86\xd0\xb0"}'
>>>
```

This should be reverted back to the way it was.

Reply:
I'm ashamed to admit that the only reason for removing `str()` because @isbm asked rather suggestively why it's there. Ie. the reason is that more attention got paid for who is saying that what was being said.

Reply:
@samilaine @terminalmage *\*ahemmm\**, I didn't really meant "please remove it and never use it again". I just thought that `salt.utils.stringutils.to_str` is more correct to use here instead of just `str`.

Reply:
Communication between humans usually fails and when it doesn't, it's an accident. By no means I tried to lay blame on anyone else, as it was yours truly who failed to comprehend that "please remove it and never use it again" wasn't what was meant.

Reply:
:wink: no problem, take it easy. The `str` is OK, but my perfectionist guts thinks that using standard utility for this would be more "kosher" here, since it does things according to the current Python version. I would leave this to @terminalmage — he is the Unicode boss here.

Reply:
`to_str()` is only necessary when it's possible that the string may contain unicode, as it will perform the necessary `.encode()` on PY2 and do nothing on PY3.

Given that we know that the format string is ascii-only, `str()` is fine here.

Reply:
@samilaine If you can get that line reverted back to what it was before, then this should be good to merge.